inngest-spring-boot-demo/src/test/java/com/inngest/springbootdemo/RateLimitedFunctionIntegrationTest.java

```diff
@@ -0,0 +1,42 @@
+package com.inngest.springbootdemo;
+
+import com.inngest.Inngest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@IntegrationTest
+@Execution(ExecutionMode.CONCURRENT)
+class RateLimitedFunctionIntegrationTest {
+    @Autowired
+    private DevServerComponent devServer;
+
+    @Autowired
+    private Inngest client;
+
+    @Test
+    void testFunctionIsRateLimited() throws Exception {
+        String event1 = InngestFunctionTestHelpers.sendEvent(client, "test/rateLimit").getIds()[0];
+        Thread.sleep(500);
+        String event2 = InngestFunctionTestHelpers.sendEvent(client, "test/rateLimit").getIds()[0];
+        Thread.sleep(500);
+        String event3 = InngestFunctionTestHelpers.sendEvent(client, "test/rateLimit").getIds()[0];
+
+        Thread.sleep(2000);
```

Comment:
Should we increase the duration similar to idempotency https://github.com/inngest/inngest-kt/pull/85?

```java
RateLimitedFunctionIntegrationTest > testFunctionIsRateLimited() FAILED
    java.lang.ArrayIndexOutOfBoundsException: 0
        at com.inngest.springbootdemo.EventRunsResponse.first(EventRunsResponse.java:14)
        at com.inngest.springbootdemo.RateLimitedFunctionIntegrationTest.testFunctionIsRateLimited(RateLimitedFunctionIntegrationTest.java:32)
```

Reply:
good idea, increased the two 2000ms waits to 4000ms

Reply:
Adjusted all the times again based on https://www.inngest.com/docs/guides/rate-limiting#how-rate-limiting-works